gavetisyang

gavetisyanca opened the pull request.

gavetisyanca committed:
Add files via upload

srwight commented on avetisyang.py:
```diff
@@ -0,0 +1,78 @@
+        
+    """
+    import pandas as pd
+	import numpy as np
```
unexpected indent

srwight commented on avetisyang.py:
```diff
@@ -0,0 +1,78 @@
+
+    # dataframe with categorical features
+    categorical_columns = [col for col in data.columns if data[col].dtype == 'object']
+	data_cat = data[categorical_columns]
```
unexpected indent

srwight commented on avetisyang.py:
```diff
@@ -0,0 +1,78 @@
+
+
+    from scipy.stats import skew
+	data_num_skew = data_num.apply(lambda x: skew(x.dropna()))
```
unexpected indent

srwight reviewed (changes requested):
Fix the indentation problems. The algorithm looks good.

gavetisyanca committed:
Update avetisyang.py

srwight reviewed (dismissed):
Looks great!

ahtouw commented on avetisyang.py:
```diff
@@ -0,0 +1,75 @@
+    import seaborn as sns
+
+
+    data = pd.read_csv("train.csv")
```
The 'df' variable passed in to the function will contain the data, no need to read the file and pass it into 'data'.

srwight replied:
Most of the function depends on the "data" variable; it might be simpler to change the function's argument to "data" rather than search out all of the instances of "data" and change them to "df"

ahtouw commented on avetisyang.py:
```diff
@@ -0,0 +1,75 @@
+	['MSSubClass','MSZoning', 'Street', 'Alley']
+
+    """
+    import pandas as pd
```
Generally good practice to import libraries outside of functions.

ahtouw reviewed (dismissed):
The 'data' variable should just be the 'df' variable that is passed in to the function on call, no need to read_csv. I don't think matplotlib or seaborn libraries need to be imported. The rest of the function looks functional.

srwight committed:
Naming Convention adjustments

Changed the name of the function and of the module to comply with naming convention:
[firstfour].py
feature_extract()

srwight commented on avet.py:
```diff
@@ -0,0 +1,75 @@
+"""
+
+
+def feature_extract(df) -> str:
```
The function returns a pd.DataFrame object, not a string. Might be good to update that (or remove it, but updating it is better. :) )

gavetisyanca committed:
Update avet.py

gavetisyanca committed:
moved libraries outside of the function

srwight commented on avet.py:
```diff
@@ -0,0 +1,78 @@
+    data_cat_dummies = pd.get_dummies(data_cat, drop_first=True)
+    # Printing number of categorical variables including dummy variables
+    # print("Categorical variables : " + str(len(data_cat_dummies.columns)))
+    return data_cat_dummies
```
Did data_num need to come to?

srwight committed:
Update avet.py

srwight reviewed (changes requested):
Are you returning everything?

srwight reviewed (approved):
This works

Hwailaluta reviewed (approved):
I think it's good.
